communication/nats/dialog/dialog_establisher.go

```diff
@@ -30,49 +30,82 @@ func NewDialogEstablisher(myIdentity identity.Identity, signer identity.Signer)
 const establisherLogPrefix = "[NATS.DialogEstablisher] "
 
 type dialogEstablisher struct {
-	myIdentity            identity.Identity
-	mySigner              identity.Signer
-	contactAddressFactory func(contact dto_discovery.Contact) (*discovery.AddressNATS, error)
+	myID               identity.Identity
+	mySigner           identity.Signer
+	peerAddressFactory func(contact dto_discovery.Contact) (*discovery.AddressNATS, error)
 }
 
-func (establisher *dialogEstablisher) CreateDialog(contact dto_discovery.Contact) (communication.Dialog, error) {
+func (establisher *dialogEstablisher) CreateDialog(
```

Comment:
This method is doing a lot of different things - it's tests are hard to understand, because it requires a lot of different mocking.
Have you though about splitting it up?

Reply:
Yes. I started splitting to smaller functions like `newCodecForPeer()` and `newSenderToPeer`.
Later they will be injectable factories